app/validators/api/v4/questionnaire_response_payload_validator.rb

```diff
@@ -0,0 +1,19 @@
+class Api::V4::QuestionnaireResponsePayloadValidator < Api::V3::PayloadValidator
+  attr_accessor(
+    :id,
+    :questionnaire_id,
+    :questionnaire_type,
+    :facility_id,
+    :user_id,
+    :content,
```

Comment:
@govindkrjoshi there are 2 kinds of validations & we feel we cannot do them as part of the sync_from_user API request
1. Invalidate presence of extra keys 
  - We cannot invalidate any extra keys that don't belong to a questionnaire layout because they might belong to older questionnaire's layout
2. Validate data type of all keys, or at-least those keys that are found inside given questionnaire layout
  - This is quite expensive performance-wise. For each record, we'll have to first fetch the layout from DB, iterate over it recursively to fetch key-type pairs and do validation for each matching key in the content.

Rather than doing this, we feel it'll be easier to just create a [Metabase Dashboard subscription](https://www.metabase.com/docs/latest/users-guide/dashboard-subscriptions) to flag anomalies in content periodically.

Reply:
> It might be worth putting the content schema in the schema.json as well as documentation.

This cannot be done because of 2 reasons:
1. QuestionnaireResponse Sync API is for all Questionnaires, not just monthly screening report. The type of questionnaire isn't present in the Sync API payload, rather it's derived based on `questionnaire_id`
2. We want to be able to add more layouts independently of code changes. Doing schema.json based validation would mean modifying code every time we're creating newer layouts.